### PR TITLE
docs: add permission tags to JSDocs

### PR DESCRIPTION
**Pull request by kitsonk**
This is part of the effort to improve our documentation of APIs.

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]
LGTM!

File: cli/dts/lib.deno.ns.d.ts [hunk 1]
```diff
@@ -2171,6 +2218,7 @@ declare namespace Deno {
    * Requires `allow-read` permission on fromPath.
    * Requires `allow-write` permission on toPath.
    *
+   * @tags allow-read
```
Comment (crowlKats) [inline cli/dts/lib.deno.ns.d.ts:2221]:
```suggestion
   * @tags allow-read, allow-read
```

File: cli/dts/lib.deno.ns.d.ts [hunk 2]
```diff
@@ -2153,6 +2199,7 @@ declare namespace Deno {
...
```
Comment (crowlKats) [inline cli/dts/lib.deno.ns.d.ts:2202]:
```suggestion
   * @tags allow-read, allow-write
```

File: cli/dts/lib.deno.ns.d.ts [hunk 3]
```diff
@@ -562,7 +562,10 @@ declare namespace Deno {
    */
   export function exit(code?: number): never;
 
-  /** @category Runtime Environment */
+  /**
+   * @tags allow-env
```
Comment (bartlomieju) [inline cli/dts/lib.deno.ns.d.ts:566]: Would it be possible to further scope this? Ie. some APIs require "full" permissions (eg. `Deno.mainModule()`), while others work fine with allow lists

**Comment by crowlKats** [comment]
some web APIs require permissions, tags should be added for those as well

File: cli/dts/lib.deno.ns.d.ts (hunk 3 above)
Comment (kitsonk) [reply cli/dts/lib.deno.ns.d.ts:566]: Any further scoping will just make the documentation unreadable, IMO.  I retained the details in the JSDoc.  This is simply to give a visual indicator and to be able to categorise things, versus being exhaustive.

**Comment by kitsonk** [comment]
> some web APIs require permissions, tags should be added for those as well

I looked for `allow` in the JSDoc and tagged those that mentioned it.  Feel free to push to this PR.

File: cli/dts/lib.deno.ns.d.ts [hunk 4]
```diff
@@ -3229,7 +3293,7 @@ declare namespace Deno {
    *
    * for await (const conn of Deno.listen({ port: 80 })) {
    *   handleHttp(conn);
-   * }
+   * }s
```
Comment (crowlKats) [inline cli/dts/lib.deno.ns.d.ts:3296]:
```suggestion
   * }
```

**Review by crowlKats — APPROVED** [review]

**Pull request merged**